tree/treeplayer/inc/TBranchProxy.h

```diff
@@ -160,10 +168,185 @@ namespace Detail {
          }
       }
 
+private:
+      friend class ROOT::Internal::TTreeReaderValueBase;
+
+      enum class EReadType {
+         kDefault,
+         kNoDirector,
+         kReadParentNoCollection,
+         kReadParentCollectionNoPointer,
+         kReadParentCollectionPointer,
+         kReadNoParentNoBranchCountCollectionPointer,
+         kReadNoParentNoBranchCountCollectionNoPointer,
+         kReadNoParentNoBranchCountNoCollection,
+         kReadNoParentBranchCountCollectionPointer,
+         kReadNoParentBranchCountCollectionNoPointer,
+         kReadNoParentBranchCountNoCollection
+      };
+
+      EReadType GetReadType() {
+         if (fParent) {
+            if (!fCollection)
+               return EReadType::kReadParentNoCollection;
+            else {
```

Comment:
The partial usage of braces makes this code confusing. Please add braces to both branches if you are going to add them.